contrib/platform/src/com/sun/jna/platform/win32/WinInet.java

```diff
@@ -0,0 +1,471 @@
+		 */
+		public int dwHitRate;
+
+		/**
+		 * Low-order portion of the file size, in bytes.
+		 */
+		public int dwSizeLow;
+
+		/**
+		 * High-order portion of the file size, in bytes.
+		 */
+		public int dwSizeHigh;
+
+		/**
+		 * FILETIME structure that contains the last modified time of this URL,
+		 * in Greenwich mean time format.
+		 */
+		public Kernel32.FILETIME LastModifiedTime;
+
+		/**
+		 * FILETIME structure that contains the expiration time of this file, in
+		 * Greenwich mean time format.
+		 */
+		public Kernel32.FILETIME ExpireTime;
+
+		/**
+		 * FILETIME structure that contains the last accessed time, in Greenwich
+		 * mean time format.
+		 */
+		public Kernel32.FILETIME LastAccessTime;
+
+		/**
+		 * FILETIME structure that contains the last time the cache was
+		 * synchronized.
+		 */
+		public Kernel32.FILETIME LastSyncTime;
+
+		/**
+		 * Pointer to a buffer that contains the header information. The buffer
+		 * occupies the memory at the end of this structure.
+		 */
+		public Pointer lpHeaderInfo;
+
+		/**
+		 * Size of the lpHeaderInfo buffer, in TCHARs.
+		 */
+		public int dwHeaderInfoSize;
+
+		/**
+		 * Pointer to a string that contains the file name extension used to
+		 * retrieve the data as a file. The string occupies the memory area at
+		 * the end of this structure.
+		 */
+		public Pointer lpszFileExtension;
+
+		/**
+		 * A union of the last two distinct fields in INTERNET_CACHE_ENTRY_INFO
+		 */
+		public UNION u;
+
+		/**
+		 * Additional data (the path and URLs mentioned previously, and more)
+		 */
+		public byte[] additional;
+
+		public INTERNET_CACHE_ENTRY_INFO(int size) {
+			additional = new byte[size];
+		}
+
+		/**
+		 * A union of the last two distinct fields in INTERNET_CACHE_ENTRY_INFO
+		 * 
+		 * <pre>
+		 * <code>
+		 *             union {
+		 *                 DWORD dwReserved;
+		 *                 DWORD dwExemptDelta;
+		 *             };</code>
+		 * </pre>
+		 */
+		public static class UNION extends Union {
+			/**
+			 * Reserved. Must be zero.
+			 */
+			public int dwReserved;
+
+			/**
+			 * Exemption time from the last accessed time, in seconds.
+			 */
+			public int dwExemptDelta;
+		}
+
+		public String getLocalFileName() {
+			return (lpszLocalFileName == null) ? "" : lpszLocalFileName.getWideString(0);
+		}
+
+		public String getSourceUrlName() {
+			return (lpszSourceUrlName == null) ? "" : lpszSourceUrlName.getWideString(0);
+		}
+		
+		@Override
+		protected List getFieldOrder() {
+			return Arrays.asList(new String[] { "dwStructSize", "lpszSourceUrlName", "lpszLocalFileName",
+					"CacheEntryType", "dwUseCount", "dwHitRate", "dwSizeLow", "dwSizeHigh", "LastModifiedTime",
+					"ExpireTime", "LastAccessTime", "LastSyncTime", "lpHeaderInfo", "dwHeaderInfoSize",
+					"lpszFileExtension", "u", "additional" });
+		}
+
+		@Override
+		public String toString() {
+			return (lpszLocalFileName == null ? "" : lpszLocalFileName.getWideString(0) + " => ")
```

Comment:
Simply call the _getLocalFileName_ and _getSourceUrlName_ methods - don't duplicate their code